From b344bcbbfda8fbe14dadc5aa4b5dfb3ced6d76e2 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 2 Oct 2014 14:35:34 +0200
Subject: [PATCH] localectl: always print warnings with log_warning() instead
 of printf()

They really should got to stderr, not stdout.
---
 src/locale/localectl.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/locale/localectl.c b/src/locale/localectl.c
index 9325d9af6d..3690f9fc89 100644
--- a/src/locale/localectl.c
+++ b/src/locale/localectl.c
@@ -117,13 +117,12 @@ static void print_overriden_variables(void) {
         for (j = 0; j < _VARIABLE_LC_MAX; j++)
                 if (variables[j]) {
                         if (print_warning) {
-                                printf("Warning: Settings on Kernel Command Line override system locale settings in /etc/locale.conf\n");
-                                printf("    Command Line: %s=%s\n", locale_variable_to_string(j), variables[j]);
+                                log_warning("Warning: Settings on kernel command line override system locale settings in /etc/locale.conf.\n"
+                                            "  Command Line: %s=%s\n", locale_variable_to_string(j), variables[j]);
 
                                 print_warning = false;
-                                continue;
-                        }
-                        printf("                  %s=%s\n", locale_variable_to_string(j), variables[j]);
+                        } else
+                                log_warning("                %s=%s\n", locale_variable_to_string(j), variables[j]);
                 }
  finish:
         for (j = 0; j < _VARIABLE_LC_MAX; j++)
